tools/mkmodel/mkmodel.c

```diff
@@ -581,6 +832,120 @@ int convert(const char *infn, const char *outfn)
             goto error;
         }
     }
+    // Convert skins
+    model->num_skins = data->skins_count;
+    model->skins = calloc(data->skins_count, sizeof(model64_skin_t));
+    for(size_t i=0; i<data->skins_count; i++) {
+        if(data->skins[i].joints_count == 0) {
+            continue;
+        }
+        if(data->skins[i].joints_count > 24) {
+            fprintf(stderr, "Error: Found %zd joints in skin %zd.\n", data->skins[i].joints_count, i);
+            fprintf(stderr, "Error: A maximum of 24 joints are allowed in a skin.\n");
+            goto error;
+        }
+        model->skins[i].num_joints = data->skins[i].joints_count;
+        model->skins[i].joints = calloc(data->skins[i].joints_count, sizeof(model64_joint_t));
+        cgltf_accessor *ibm_accessor = data->skins[i].inverse_bind_matrices;
+        float *ibm_buffer = NULL;
+        if(ibm_accessor) {
+            size_t num_components = cgltf_num_components(ibm_accessor->type);
+            size_t num_values = num_components * ibm_accessor->count;
+            ibm_buffer = malloc(sizeof(float) * num_values);
+
+            // Convert all data to floats (because cgltf provides this very convenient function)
+            // TODO: More sophisticated conversion that doesn't always use floats as intermediate values
+            //       Might not be worth it since the majority of tools will probably only export floats anyway?
+            if (cgltf_accessor_unpack_floats(ibm_accessor, ibm_buffer, num_values) == 0) {
+                fprintf(stderr, "Error: failed reading inverse bind matrices.\n");
+                free(ibm_buffer);
+                goto error;
+            }
+        }
+        
+        for(uint32_t j=0; j<model->skins[i].num_joints; j++) {
+            model->skins[i].joints[j].node_idx = cgltf_node_index(data, data->skins[i].joints[j]);
+            if(ibm_buffer) {
+                memcpy(model->skins[i].joints[j].inverse_bind_mtx, &ibm_buffer[j*16], sizeof(float)*16);
+            } else {
+                model->skins[i].joints[j].inverse_bind_mtx[0] = 1.0f;
+                model->skins[i].joints[j].inverse_bind_mtx[5] = 1.0f;
+                model->skins[i].joints[j].inverse_bind_mtx[10] = 1.0f;
+                model->skins[i].joints[j].inverse_bind_mtx[15] = 1.0f;
+            }
+        }
+        if(ibm_buffer) {
+            free(ibm_buffer);
+        }
+    }
+    // Convert nodes
+    model->num_nodes = data->nodes_count;
+    // Add extra node if scene has multiple root nodes
+    if(data->scene->nodes_count > 1) {
+        model->num_nodes++;
+    }
+    model->nodes = calloc(model->num_nodes, sizeof(model64_node_t));
```

Comment:
Probably shouldn't allocate if there are no nodes (I know that would error out atm anyway, but see my other comment on that)